### PR TITLE
[SPARK-48629] Migrate the residual code to structured logging framework

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to migrate the `residual code` to structured logging framework.

### Why are the changes needed?
When I reviewed the spark code, I found that some logs in the some module were not fully migrated to the structured logging framework, let's to complete unfinished work.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
In addition, I have also found similar issues in the `core` module, but currently there seems to be a PR https://issues.apache.org/jira/browse/SPARK-47579 still in progress, so to prevent conflicts, I skipped it.

**Comment by gengliangwang** [comment]
@panbingkun Could you update this one?

**Comment by panbingkun** [comment]
> @panbingkun Could you update this one?

Okay, sure, let me do it.

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala [hunk 1]
```diff
@@ -869,7 +868,7 @@ private[spark] class ApplicationMaster(
           }
         } else {
           logError(log"Application Master lost connection with driver! Shutting down. " +
-            log"${MDC(LogKeys.REMOTE_ADDRESS, remoteAddress)}")
+            log"${MDC(LogKeys.HOST_PORT, remoteAddress)}")
```
Comment (gengliangwang) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:871]: why renaming this one?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala [hunk 2]
```diff
@@ -373,8 +372,7 @@ abstract class StreamExecution(
           case _ => None
         }
 
-        logError(log"Query ${MDC(LogKeys.PRETTY_ID_STRING, prettyIdString)} " +
-          log"terminated with error", e)
+        logError(log"Query ${MDC(QUERY_ID, prettyIdString)} terminated with error", e)
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala:375]: Why renaming this one?

**Comment by panbingkun** [comment]
Let me double check myself.

File: common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala [hunk 3]
```diff
@@ -299,9 +301,8 @@ private[spark] object LogKeys {
   case object HIVE_OPERATION_STATE extends LogKey
   case object HIVE_OPERATION_TYPE extends LogKey
   case object HOST extends LogKey
+  case object HOSTS extends LogKey
   case object HOST_LOCAL_BLOCKS_SIZE extends LogKey
-  case object HOST_NAME extends LogKey
```
Comment (panbingkun) [inline common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala:303]: Let's unify `HOST_NAME` to `HOST` and remove `HOST_NAME`

File: common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala [hunk 4]
```diff
@@ -299,9 +301,8 @@ private[spark] object LogKeys {
...
-  case object HOST_NAMES extends LogKey
```
Comment (panbingkun) [inline common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala:304]: Let's unify `HOST_NAMES` to `HOSTS` and remove `HOST_NAMES`

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 5]
```diff
@@ -83,7 +83,7 @@ private[spark] class Executor(
   extends Logging {
 
   logInfo(log"Starting executor ID ${LogMDC(LogKeys.EXECUTOR_ID, executorId)}" +
-    log" on host ${LogMDC(HOST_NAME, executorHostname)}")
+    log" on host ${LogMDC(HOST, executorHostname)}")
```
Comment (panbingkun) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala:86]: Let's unify it as `HOST`, the general rule is
HOST: Only `host name`, not including `port`, eg: host_name
Port: Just the `port`, eg: 10240
HOST_PORT: includes both host name and port, eg: `host_name:10240`

Comment (panbingkun) [reply core/src/main/scala/org/apache/spark/executor/Executor.scala:86]: https://github.com/apache/spark/blob/e972daec30eb34e9352f37e9ac74db77a895d679/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala#L765

File: core/src/main/scala/org/apache/spark/deploy/master/ui/MasterWebUI.scala [hunk 6]
```diff
@@ -79,7 +79,7 @@ class MasterWebUI(
           } else {
             val removedWorkers = masterEndpointRef.askSync[Integer](
               DecommissionWorkersOnHosts(hostnames))
-            logInfo(log"Decommissioning of hosts ${MDC(HOST_NAMES, hostnames)}" +
+            logInfo(log"Decommissioning of hosts ${MDC(HOSTS, hostnames)}" +
```
Comment (panbingkun) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/MasterWebUI.scala:82]: Let's unify it as `HOSTS`, there is a detailed explanation below.

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListenerBus.scala [hunk 7]
```diff
@@ -145,13 +145,11 @@ class StreamingQueryListenerBus(sparkSession: SparkSession) extends Logging {
           case t: QueryTerminatedEvent =>
             listener.onQueryTerminated(t)
           case _ =>
-            logWarning(
-              log"Unknown StreamingQueryListener event: " +
-                log"${MDC(LogKeys.EVENT, event)}")
+            logWarning(log"Unknown StreamingQueryListener event: ${MDC(LogKeys.EVENT, event)}")
```
Comment (panbingkun) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListenerBus.scala:148]: Make our code format look pretty

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala [hunk 8]
```diff
@@ -176,8 +176,7 @@ class StateStoreChangelogWriterV1(
     } catch {
       case e: Throwable =>
         abort()
-        logError(log"Fail to commit changelog file ${MDC(LogKeys.FILE_NAME, file)} " +
```
Comment (panbingkun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala:179]: Let's call it `PATH`

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala (hunk 1 above)
Comment (panbingkun) [reply resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:871]: I remember there was a PR before that had already unified `REMOTE_ADDRESS` as `HOST_PORT`

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala [hunk 9]
```diff
@@ -527,10 +527,9 @@ private[spark] class ApplicationMaster(
       userClassThread.join()
     } catch {
       case e: SparkException if e.getCause().isInstanceOf[TimeoutException] =>
-        logError(
-          log"""SparkContext did not initialize after waiting for
-          |${MDC(LogKeys.TIMEOUT, totalWaitTime)} ms.
-          | Please check earlier log output for errors. Failing the application.""".stripMargin)
+        logError(log"SparkContext did not initialize after waiting for " +
```
Comment (panbingkun) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:530]: Make the code format pretty

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala (hunk 2 above)
Comment (panbingkun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala:375]: Let's restore it, I misread it

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/Operation.java [hunk 10]
```diff
@@ -290,7 +290,7 @@ protected void cleanupOperationLog() {
       if (operationLog == null) {
         LOG.error("Operation [ {} ] logging is enabled, " +
           "but its OperationLog object cannot be found.",
-          MDC.of(LogKeys.OPERATION_HANDLE_IDENTIFIER$.MODULE$, opHandle.getHandleIdentifier()));
```
Comment (panbingkun) [inline sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/Operation.java:293]: Let's abbreviate `IDENTIFIER` as `ID`

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetFunctionsOperation.scala [hunk 11]
```diff
@@ -51,9 +51,11 @@ private[hive] class SparkGetFunctionsOperation(
 
   override def runInternal(): Unit = {
     // Do not change cmdStr. It's used for Hive auditing and authorization.
-    val cmdStr = s"catalog : $catalogName, schemaPattern : $schemaName"
-    val logMsg = s"Listing functions '$cmdStr, functionName : $functionName'"
-    logInfo(log"${MDC(LogKeys.MESSAGE, logMsg)} with ${MDC(LogKeys.STATEMENT_ID, statementId)}")
+    val cmdMDC = log"catalog : ${MDC(LogKeys.CATALOG_NAME, catalogName)}, " +
```
Comment (panbingkun) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetFunctionsOperation.scala:54]: Let's make them more reasonable

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetFunctionsOperation.scala [hunk 12]
```diff
@@ -51,9 +51,11 @@ private[hive] class SparkGetFunctionsOperation(
 
   override def runInternal(): Unit = {
     // Do not change cmdStr. It's used for Hive auditing and authorization.
-    val cmdStr = s"catalog : $catalogName, schemaPattern : $schemaName"
-    val logMsg = s"Listing functions '$cmdStr, functionName : $functionName'"
-    logInfo(log"${MDC(LogKeys.MESSAGE, logMsg)} with ${MDC(LogKeys.STATEMENT_ID, statementId)}")
```
Comment (panbingkun) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetFunctionsOperation.scala:56]: expand `logMsg`

**Comment by panbingkun** [comment]
@gengliangwang 
It's ready.

**Review by gengliangwang — APPROVED** [review]
Thank you for your meticulous work!

**Comment by gengliangwang** [comment]
Merging to master

**Comment by panbingkun** [comment]
> Thank you for your meticulous work!

Thank you for your patient review. ❤️

**Pull request closed without merging**